secret/secret_test.go

```diff
@@ -333,17 +310,17 @@ func TestGenerateSecretsWithNoSecrets(t *testing.T) {
 			},
 		}
 
-		secrets := v1.Secret{Data: map[string][]byte{}}
-		updates := v1.Secret{Data: map[string][]byte{}}
+		secrets := &v1.Secret{Data: map[string][]byte{}}
+		updates := &v1.Secret{Data: map[string][]byte{}}
 
 		var mockSecrets MockSecrets
 		generateSSLCerts = mockSecrets.GenerateSSLCerts
 		passGenerate = mockSecrets.PassGenerate
-		mockSecrets.On("GenerateSSLCerts", &secrets, &updates).Return(false)
-		mockSecrets.On("PassGenerate", map[string][]byte{}, map[string][]byte{}, "clean").Return(false)
-		dirty := GenerateSecrets(manifest, &secrets, &updates)
-		assert.False(dirty)
-		mockSecrets.AssertCalled(t, "PassGenerate", map[string][]byte{}, map[string][]byte{}, "clean")
+		mockSecrets.On("GenerateSSLCerts", secrets, updates)
+		mockSecrets.On("PassGenerate", secrets, updates, "clean")
+
+		GenerateSecrets(manifest, secrets, updates)
+		mockSecrets.AssertCalled(t, "PassGenerate", secrets, updates, "clean")
```

Comment:
Assert that it's _not_ dirty? (Also the next one; similar for more removal of asserting true/false.)